### ISSUE TITLE
Serialization profiling 

### ISSUE THREAD

**Issue by hackaugusto** [brainbot-com/raiden#19]
Measure the impact of the serialization on performance.

Related issue: #14

**Comment by hackaugusto** [comment]
https://gist.github.com/hackaugusto/34d697209cb83a075507

**Comment by hackaugusto** [comment]
11b1bf71e4945c42b34cbc5ef8818c66d04886aa adds a test script for the serialization speed, these are the results from my machine:

```
# python ./raiden/raiden/tests/speed_decoding.py
Ack: encode 13.7116439342 decode 11.8796238899
Ping: encode 13.6333608627 decode 140.372318029
SecretRequest: encode 13.083065033 decode 140.492857933
Transfer: encode 30.6614868641 decode 153.627314091
LockedTransfer: encode 43.1515641212 decode 163.61800909
MediateTranfer: encode 52.5549809933 decode 171.038762093
CancelTransfer: encode 43.4540059566 decode 162.903224945
TransferTimeout: encode 15.7039411068 decode 141.257529974
ConfirmTransfer: encode 13.0377030373 decode 140.529330969
```

**Comment by hackaugusto** [comment]
Ack is the only message that is not signed, for the signed messages the decoding time is dominated by the pubkey recovery.

Here is another run of the `speed_decoding.py`:

```
Ack: encode 13.6720991135 decode 13.6039540768
Ping: encode 13.6513791084 decode 142.310990095
SecretRequest: encode 12.9184000492 decode 136.955137968
Transfer: encode 29.5317671299 decode 150.243940115
LockedTransfer: encode 42.1850368977 decode 162.588837147
MediateTranfer: encode 51.2585670948 decode 169.336026907
CancelTransfer: encode 42.0469717979 decode 159.478933096
TransferTimeout: encode 15.1263501644 decode 138.429334879
ConfirmTransfer: encode 12.5078909397 decode 137.260711193
```

Here is a run with pubkey recovery disabled:

```
Ack: encode 13.8763689995 decode 13.5622711182
Ping: encode 13.5759370327 decode 13.2449560165
SecretRequest: encode 12.6620140076 decode 11.9292969704
Transfer: encode 30.0450150967 decode 24.8050620556
LockedTransfer: encode 42.7343661785 decode 35.0820598602
MediateTranfer: encode 53.5386459827 decode 42.9855029583
CancelTransfer: encode 43.3911540508 decode 34.603397131
TransferTimeout: encode 15.5850450993 decode 14.1954929829
ConfirmTransfer: encode 13.003538847 decode 12.3270828724
```

The method `_recover_sender` use two functions, `sha3` and `c_ecdsa_recover_compact`, these are theirs timings:

```
>>> import timeit, random
>>> from raiden.encoding import c_ecdsa_sign_compact, c_ecdsa_recover_compact
>>> priv = ''.join(chr(random.randint(0, 255)) for i in range(32))
>>> msg = ''.join(chr(random.randint(0, 255)) for i in range(32))
>>> sig = c_ecdsa_sign_compact(msg, priv)
>>> def test(): c_ecdsa_recover_compact(msg, sig)
... 
>>> timeit.timeit(test)
92.00966811180115
>>> timeit.timeit('sha3(d)', 'from raiden.encoding import sha3; d = "x" * 250')
14.392619848251343
```

**Comment by heikoheiko** [comment]
100μs x 10.000 == 1s. as expected.

**Comment by hackaugusto** [comment]
The fixed length format #20  is a bit faster (considering encoding/decoding only, without signing):
```
Ack: encode 6.29723906517 decode 5.89036202431
Ping: encode 9.25081014633 decode 9.61398696899
SecretRequest: encode 6.93648386002 decode 7.03274202347
Transfer: encode 24.239577055 decode 20.0348110199
LockedTransfer: encode 32.7933149338 decode 27.2950229645
MediatedTranfer: encode 39.0207471848 decode 33.5120608807
CancelTransfer: encode 32.5260181427 decode 27.2790400982
TransferTimeout: encode 8.62257122993 decode 8.29779005051
ConfirmTransfer: encode 7.05855512619 decode 7.13217115402
```
